File: bindings/haskell/examples/basic.hs
module Main where

import Antech.Kdf
import qualified Data.ByteString.Char8 as B8

main :: IO ()
main = do
  stored <- hash (B8.pack "correct_horse_battery_staple")
  ok <- verify (B8.pack "correct_horse_battery_staple") stored
  if not ok then error "verify failed" else pure ()
  cfg0 <- configDefault
  let cfg = cfg0 { memoryKib = 1024 }
  custom <- hashWithConfig (B8.pack "pw") cfg
  needs <- needsRehash custom
  putStrLn ("needs_rehash " ++ show needs)
  putStrLn stored

File: bindings/zig/examples/basic.zig
const std = @import("std");
const antech = @import("antech_kdf");

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    defer _ = gpa.deinit();
    const allocator = gpa.allocator();

    const stored = try antech.hash(allocator, "correct_horse_battery_staple");
    defer allocator.free(stored);
    const stored_z = try allocator.dupeZ(u8, stored);
    defer allocator.free(stored_z);

    if (!(try antech.verify("correct_horse_battery_staple", stored_z))) {
        return error.VerifyFailed;
    }

    var cfg = try antech.Config.default();
    cfg.memory_kib = 1024;
    const custom = try antech.hashWithConfig(allocator, "pw", cfg);
    defer allocator.free(custom);
    const custom_z = try allocator.dupeZ(u8, custom);
    defer allocator.free(custom_z);

    std.debug.print("needs_rehash {}\n", .{try antech.needsRehash(custom_z)});
    std.debug.print("{s}\n", .{stored});
}
